### DISCUSSION TITLE
SQL Server database - message transmission errors?

### DISCUSSION THREAD

**Discussion by sfmans** [nextgenhealthcare/connect#4771] [Q&A] [answered]
We use SQL Server as our Mirth back-end database and thanks to https://appsbyaaron.wordpress.com/2017/10/31/mirth-database-tables/ have been able to identify many areas of the data. 

However I cannot find the current stats for errors on channels; the ERROR column in the table D_Ms for that channel would seem to be the most likely place but that shows 0, despite that channel currently showing 977 errors in the Mirth GUI.  

I would like to incorporate that statistic into our monitoring, which already successfully shows some Mirth stats derived from the SQL database. All pointers gratefully received

**Comment by pacmano1** [comment, marked as answer]
table d_ms#

Although mirth's API provides endpoints to get stats like that.  

Api swagger docs are at:  
```html 
https://<yourservername>:8443/api/
 ```

Assuming you are logged in:

```shell
curl -X GET "https://<yourservername>:8443/api/channels/1bd72bb2-42d2-4300-91ef-d066cc7b7a66/statistics" -H "accept: application/json"
```
Returns:
```json
{
  "channelStatistics": {
    "serverId": "2367ad66-8ed0-4c27-84c1-cd6742d61650",
    "channelId": "1bd72bb2-42d2-4300-91ef-d066cc7b7a66",
    "received": 629153,
    "sent": 623933,
    "error": 5,
    "filtered": 0,
    "queued": 0
  }
}
```

To get a full list:
```shell
curl -X GET "https://<yourservername>:8443/api/channels/statistics?includeUndeployed=true&aggregateStats=false" -H "accept: application/xml"
```
